esp/services/ws_ecl/ws_ecl_wuinfo.cpp

```diff
@@ -209,11 +209,39 @@ bool WsEclWuInfo::getWsResource(const char *name, StringBuffer &out)
         IArrayOf<IPropertyTree> parts;
         if (fields.length())
         {
+            bool addedAdditionalFields = false;
+            unsigned offset = 1;
             ForEachItemIn(i, fields)
             {
-                Owned<IConstWUResult> var = wu->getVariableByName(fields.item(i));
-                if (var)
-                    appendVariableParmInfo(parts, resultSetFactory, *var, i+1);
+                const char *name = fields.item(i);
+                if (!addedAdditionalFields && streq(name, "*")) //name of '*' means insert all unspecified fields here, can only be used once
+                {
+                    addedAdditionalFields=true;
+                    Owned<IConstWUResultIterator> vars = &ensureWorkUnit()->getVariables();
+                    ForEach(*vars)
+                    {
+                        SCMStringBuffer s;
+                        IConstWUResult &var = vars->query();
+                        if (NotFound == fields.find(var.getResultName(s).str()))
+                        {
+                            SCMStringBuffer fieldSeq;
+                            var.getResultFieldOpt("sequence", fieldSeq);
+                            int seq = fieldSeq.length() ? atoi(fieldSeq.str()) : 0;
+                            if (seq < 0)
+                                seq = 0;
+                            seq = seq + i + 1;
```

Comment:
No the fieldSeq is being adapted to work a bit differently here.  When #WebService is not in use the fieldSeq is a relative position to all the other fields.  I.e. for managing over time you may use sparse numbering..   100, 150, 200, 300.  So I can easily insert a new entry somewhere.

What I am doing here allows you to put those in your #WebService field list, so I respect that numbering but make sure it fits where it belongs in our list.  Basically things inserted here in the list use fieldSeq only to determine there order relative to each other, but not fields named explicitly in the list.

So WebService(Fields('abc', *, def'));
anything added by '*' uses fieldSeq, but must be inserted between abc and def which do not use fieldSeq.